generators/app/templates/.gitignore

```diff
@@ -1,35 +1,49 @@
-################################################
-# macOS
-#################################################
-*.DS_Store
-.AppleDouble
-.LSOverride
+# System
+# ---------------------------
+.DS_Store
+Thumbs.db
 
-# Icon must end with two \r
-Icon
+# Config
+# ---------------------------
+*.esproj
+*.sublime-project
+*.sublime-workspace
+*.tmproj
+*.tmproject
+.env
+.env.php
+.idea
+.project
+.settings
 
-# Thumbnails
-._*
+# Compiled Files
+# ---------------------------
+/dist/
 
-# Files that might appear in the root of a volume
-.DocumentRevisions-V100
-.fseventsd
-.Spotlight-V100
-.TemporaryItems
-.Trashes
-.VolumeIcon.icns
-.com.apple.timemachine.donotpresent
+# Craft
+# ---------------------------
+/craft/plugins/*
+/craft/rev-manifest.json
+/craft/storage/*
 
-# Directories potentially created on remote AFP share
-.AppleDB
-.AppleDesktop
-Network Trash Folder
-Temporary Items
-.apdisk
+# Craft Public
+# ---------------------------
+/public/*
```

Comment:
I think this is okay. Looking at a few projects it seems like we're stilling going to end up with a handful of files in this project that we never want ignored:

```
apple-touch-icon.png
favicon-16x16.png
favicon-32x32.png
favicon.ico
robots.txt
```

Is there a way to wildcard a filename for non-ignoring? e.g. `!/public/favicon*`